kserve/odh-overlays/kustomization.yaml

```diff
@@ -0,0 +1,49 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - ../kserve-built
+  - network-policies.yaml
+
+patches:
+  - path: ./kserve-controller-manager-patch.yaml
+  - path: ./inferenceservice-config-patch.yaml
+  - path: ./kserve-webhook-certificate-patch.yaml
+
+  # Remove invalid CA bundles
+  - patch: |-
+      - op: remove
+        path: "/spec/conversion/webhook/clientConfig/caBundle"
+    target:
+      kind: CustomResourceDefinition
+      name: inferenceservices.serving.kserve.io
+  - patch: |-
+      - op: remove
+        path: "/webhooks/0/clientConfig/caBundle"
+      - op: remove
+        path: "/webhooks/1/clientConfig/caBundle"
+    target:
+      kind: MutatingWebhookConfiguration
+  - patch: |-
+      - op: remove
+        path: "/webhooks/0/clientConfig/caBundle"
+    target:
+      kind: ValidatingWebhookConfiguration
+
+  # Inject CA bundle into webhooks
+  - patch: |-
+      - op: add
+        path: "/metadata/annotations/cert-manager.io~1inject-ca-from"
+        value: "opendatahub/serving-cert"
```

Comment:
@israel-hdez any idea if we can use the namespace in the base/kustomization.yaml here somehow?

Reply:
Hmm, I don't know.

A quick google search lead me to this: https://github.com/kubernetes-sigs/kustomize/issues/603#issue-388571646. The `objref` targets a `Namespace`, but if there isn't one in our case, I think we may use any existent object.

It's just a pointer. I don't know if it is good for our case.

Reply:
Maybe @bartoszmajsak can recommend something here. I think he dealt with something similar.

Reply:
@VaishnaviHire @LaVLaS @anishasthana do you have any idea/experience with a similar use case?
We need to inject the namespace value as parameter in these places
https://github.com/skonto/odh-manifests/commit/68210020873d31af27859f336d8ac8725e32e52c

It is possible with a recent version of kustomize but this is not really an option at this stage so I guess if you solved this problem in the past with some different workaround.
We can probably hardcoded a similar "pattern" in the operator codebase itself (like `Replace(rawResource, "$(NAMESPACE)", r.namespace)`) but it is quite a hack :)

Reply:
From the top of my head, if you don't want to change / hack anything in the source code of the operator, you could define a k8s `batch/v1 Job` as part of your kustomize manifests in which you can apply this patch using regular `kubectl patch`. At this point, you will know your target namespace e.g. `NAMESPACE=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)` and use envsubst in the Job script to replace $NAMESPACE in the patch before applying.